projects/components/src/table/table.component.ts

```diff
@@ -606,9 +684,11 @@ export class TableComponent
       if (allRowsSelected) {
         this.dataSource?.selectAllRows();
         this.selections = this.dataSource?.getAllRows();
+        this.allRowsSelectionChecked = true;
```

Comment:
Nitpick: It could be informative to add a doc comment explaining if "all rows" refers to the current page, or the entirety of the data.

Reply:
Added with -> https://github.com/hypertrace/hypertrace-ui/pull/2170/commits/999ce0ea53b5f162b1e2ef41f3eeb1e1250ca4ad